test_libs/pyspec/eth2spec/test/phase_0/epoch_processing/test_process_justification_and_finalization.py

```diff
@@ -0,0 +1,74 @@
+from eth2spec.test.context import spec_state_test, with_all_phases
+from eth2spec.test.phase_0.epoch_processing.run_epoch_process_base import (
+    run_epoch_processing_with
+)
+
+
+def run_process_just_and_fin(spec, state):
+    yield from run_epoch_processing_with(spec, state, 'process_justification_and_finalization')
+
+
+def get_committee_size(spec, state, slot):
+    epoch = spec.slot_to_epoch(slot)
+    epoch_start_shard = spec.get_epoch_start_shard(state, epoch)
+    committees_per_slot = spec.get_epoch_committee_count(state, epoch) // spec.SLOTS_PER_EPOCH
+    shard = (epoch_start_shard + committees_per_slot * (slot % spec.SLOTS_PER_EPOCH)) % spec.SHARD_COUNT
+    committee_index = (shard + spec.SHARD_COUNT - spec.get_epoch_start_shard(state, epoch)) % spec.SHARD_COUNT
+    committee_count = spec.get_epoch_committee_count(state, epoch)
+    indices = spec.get_active_validator_indices(state, epoch)
+    start = (len(indices) * committee_index) // committee_count
+    end = (len(indices) * (committee_index + 1)) // committee_count
+    size = end - start
+    return size
+
+
+def add_mock_attestations(spec, state, target_epoch, att_count, att_ratio):
+    # we must be at the end of the epoch
+    assert (state.slot + 1) % spec.SLOTS_PER_EPOCH == 0
+
+    previous_epoch = spec.get_previous_epoch(state)
+    current_epoch = spec.get_current_epoch(state)
+
+    if current_epoch == target_epoch:
+        attestations = state.current_epoch_attestations
+    elif previous_epoch == target_epoch:
+        attestations = state.previous_epoch_attestations
```

Comment:
Are we assuming these `attestation` lists are empty at the start of this method? If so, maybe add another sanity check assert at the top.

Reply:
I think calling the function multiple times is a feature. But yes, may need other changes to do that well. Will see how far we can test with latest version of the helper func.